client/marklogicUnitTestClient.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import {
+    ExtensionContext, TextEditor, WorkspaceConfiguration
+} from 'vscode';
+
+import { ClientContext, MlClientParameters, requestMarkLogicUnitTest } from './marklogicClient';
+import { buildMlxprsErrorFromError, MlxprsError } from './mlxprsErrorBuilder';
+import { MlxprsErrorReporter } from './mlxprsErrorReporter';
+import { MlxprsWebViewProvider } from './mlxprsWebViewProvider';
+
+export class MarkLogicUnitTestClient {
+    static mlxprsWebViewProvider: MlxprsWebViewProvider = null;
+
+    public static registerMlxprsResultsViewProvider(mlxprsWebViewProvider: MlxprsWebViewProvider) {
+        MarkLogicUnitTestClient.mlxprsWebViewProvider = mlxprsWebViewProvider;
+    }
+
+    private extensionContext: ExtensionContext;
+
+    public constructor(context: ExtensionContext) {
+        this.extensionContext = context;
+    }
+
+    private newMarklogicUnitTestClient(cfg: WorkspaceConfiguration): ClientContext {
+        const configParams: Record<string, unknown> = {
+            host: String(cfg.get('marklogic.host')),
+            user: String(cfg.get('marklogic.username')),
+            pwd: String(cfg.get('marklogic.password')),
+            port: Number(cfg.get('marklogic.unitTestPort')),
+            contentDb: null,
+            modulesDb: null,
+            authType: String(cfg.get('marklogic.authType')).toUpperCase(),
+            ssl: Boolean(cfg.get('marklogic.ssl')),
+            pathToCa: String(cfg.get('marklogic.pathToCa') || ''),
+            rejectUnauthorized: Boolean(cfg.get('marklogic.rejectUnauthorized'))
+        };
+        const newParams = new MlClientParameters(configParams);
+        return new ClientContext(newParams);
+    }
+
+    public async sendTestFile(
+        cfg: WorkspaceConfiguration, editor: TextEditor
+    ): Promise<void> {
+        const splitString = 'test/suites/';
+        const filePath = editor.document.uri.path;
+        const splitLocation = filePath.lastIndexOf(splitString) + splitString.length;
+        if (splitLocation > splitString.length) {
```

Comment:
If this is not the case - which I assume could happen if I run this command on a file that is not under "test/suites" - then nothing happens, correct? I think we'd want a message explaining why nothing happened - e.g. "Unable to run test; please ensure the file is a marklogic-unit-test module located in a test suite directory". (I'm also wondering if that will work on Windows because of the use of forward slashes).

Reply:
Added in code to get the OS path separator and it produces an error message (as suggested) when the expected path does not exist.